### ISSUE TITLE
vue/require-explicit-emits flags for TS events imported from elsewhere (using betterDefine)

### ISSUE THREAD

**Issue by richardtallent-erm** [vuejs/eslint-plugin-vue#2029]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 8.26.0 (latest)
- **eslint-plugin-vue version:** 9.7.0 (latest)
- **Node version:** 8.13.2
- **Operating System:** Windows x64

**Please show your full configuration:**
```js
const ELEMENTS_CONTENT_WRAP_ALLOWED = ["div", "pre", "textarea", "p", "b", "i", "a", "u", "s", "em", "strong", "template", "router-link"]

module.exports = {
	plugins: ["vuejs-accessibility", "vue"],
	extends: ["eslint:recommended", "plugin:vue/vue3-recommended", "plugin:vuejs-accessibility/recommended", "plugin:@typescript-eslint/recommended", "prettier"],
	env: {
		node: true,
	},
	root: true,
	parser: "vue-eslint-parser",
	parserOptions: {
		ecmaVersion: "latest",
		parser: "@typescript-eslint/parser",
	},
	rules: {
		indent: [0, "tab"],
		quotes: [
			"error",
			"double",
			{
				avoidEscape: true,
			},
		],
		"space-before-function-paren": [
			"warn",
			{
				anonymous: "always",
				named: "never",
				asyncArrow: "always",
			},
		],
		"arrow-parens": 0,
		"no-console": "warn",
		"eol-last": "off",
		"vue/no-v-html": 0,
		"vue/max-attributes-per-line": "off",
		"vue/html-closing-bracket-newline": 0,
		"vue/html-self-closing": [
			"error",
			{
				html: {
					void: "always",
					normal: "always",
					component: "always",
				},
				svg: "any",
				math: "always",
			},
		],
		"vue/no-mutating-props": "warn",
		"vue/html-indent": 0,
		"vue/multiline-html-element-content-newline": [
			1,
			{
				ignoreWhenEmpty: true,
				ignores: ELEMENTS_CONTENT_WRAP_ALLOWED,
				allowEmptyLines: false,
			},
		],
		"no-unused-vars": "off",
		"@typescript-eslint/no-unused-vars": ["warn"],
		"vue/first-attribute-linebreak": [
			"error",
			{
				singleline: "ignore",
				multiline: "below",
			},
		],
	},
}

```

**What did you do?**	

I'm using the [betterDefine plugin](https://vue-macros.sxzz.moe/features/better-define.html), which allows using `defineProp()` and `defineEmits()` with *imported* TypeScript interfaces. Volar and Vite are happy with this, but eslint doesn't "see" the events defined in the imported interface, so it flags them.

```FieldProps.ts
export interface IFieldEmits {
	(e: "blur", value: FocusEvent)
}
```

```vue
import type { IFieldEmits } from "./FieldProps"
const emit = defineEmits<IFieldEmits>()
...
const onBlur = (event: FocusEvent) => emit("blur", event)
```

**What did you expect to happen?**

No warnings.

**What actually happened?**

The string `"blur"` has a warning:

```
The "blur" event has been triggered but not declared on `defineEmits`.eslint[vue/require-explicit-emits](https://eslint.vuejs.org/rules/require-explicit-emits.html)
```

**Repository to reproduce this issue**

I'll follow up on this, it's going to take some time to create something small enough, this is a large app.

**Comment by ota-meshi** [comment]
Your usage is not really supported by vue compiler.

https://github.com/vuejs/core/issues/4294#issuecomment-896082769

**Comment by AmBeta** [comment]
As vue has supported this kind of usage from version 3.3, this feature should be reconsidered.
